src/resourcepack-packager/src/utils/blockbenchUtils.ts

```diff
@@ -0,0 +1,52 @@
+import parseDataUrl, { DataUrl } from 'parse-data-url';
+import { Animation, Manifest } from '../types';
+
+export function obtainBase64PackIcon(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    // @ts-ignore
+    Screencam.cleanCanvas({ width: 100, height: 100 }, (dataUrl: string) => {
+      const parsedDataUrl = parseDataUrl(dataUrl);
+
+      if (!parsedDataUrl)
+        reject(new Error('unable to parse screenshot data url'));
+
+      resolve((parsedDataUrl as DataUrl).data);
+    });
+  });
+}
+
+export function buildManifest(
+  description: string,
+  pack_format: number
+): Manifest {
+  return {
+    pack: {
+      description,
+      pack_format
+    }
+  };
+}
+
+export function compileJavaBlockCodec(
+  enableCredits: boolean = false,
+  credits: string = ''
+) {
+  // @ts-ignore
+  const data = JSON.parse(Codecs.java_block.compile());
```

Comment:
You can use `Codecs.java_block.compile({raw: true})` so that you don't have to parse it again.